vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1692,6 +1693,15 @@ def generate_draft_token_ids(
                 draft_token_ids.append(drafter_output.tolist())
         return draft_token_ids
 
+    def update_config(self, overrides: dict[str, Any]) -> None:
+        for config_name, config_overrides in overrides.items():
+            try:
+                config = getattr(self, config_name)
+            except AttributeError as exc:
+                raise ValueError(f"Unknown config {config_name}") from exc
+            new_config = dataclasses.replace(config, **config_overrides)
+            setattr(self, config_name, new_config)
```

Comment:
ideally we shouldn't override the default setattr function for this class, but to be future proof, we should use `object.__setattr__` (setattr(self, k,v) is syntatic sugar for  self.__setattr__(k,v) here)

Reply:
I searched vllm codebase and find everywhere uses `setattr`. I'll leave it to unit test to catch any future change